doc/ug/getting_started_sw_testing.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Test Software
+---
+
+## Prerequisites
+
+_Make sure you followed the install instructions to [prepare the system]({{< relref "install_instructions#system-preparation" >}}) and to install the [software development tools]({{< relref "doc/ug/install_instructions#software-development" >}}) and [Verilator]({{< relref "install_instructions#verilator" >}})._
+
+## Overview
+
+The OpenTitan software tests fall into two main categories:
+
+* Unit tests
+* Smoke tests
+
+Unit tests are designed to run on the host so they are quick to build and run.
+Writing unit tests is a good way to provide high levels of test coverage for a library without significantly impacting the runtime of the continuous integration tests.
```

Comment:
I wouldn't distinguish primarily based on the runtime of the test, but on what the test covers.

Our unit tests test currently down to the memory access/register boundary. We could define the unit boundaries differently for different tests, but I don't think we have a test currently that mocks something other than register accesses.

The benefit of mocking is that these tests can be run without actually modeling any OpenTitan device and hence on the host cpu directly.

The on-device tests allow us to test beyond the register read/write boundary and actually include the actual device behavior in the test.